portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,22 +20,36 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.*;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.PreparedQuery;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
-  ArrayList<String> comments = new ArrayList<String>() {};
   
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    //response.setContentType("text/html;");
-    //response.getWriter().println("<h1>A work in progress...</h1>");
+    //Query to datastore to retrieve comments
+    PreparedQuery results = datastore.prepare(
+        new Query("Comment").addSort("timestamp", SortDirection.DESCENDING)
+    );
 
-    //Responding with json-converted ArrayList
-    String json = convertToJson(comments);
+    ArrayList<String> comments = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+      String content = (String) entity.getProperty("content");
+      if (comments.size() < 8) comments.add(content);
```

Comment:
why is 8 hard coded here? what does it refer to?

Reply:
Limits number of comments shown to 8 as stated in the last commit for 'week 2 store'. I struggled to find a way to update this with a button, so I will be looking more into it during this week.